x-pack/plugins/fleet/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -117,6 +120,290 @@ export const installTransform = async (
+    await Promise.all(
+      destinationIndexTemplates
+        .map((destinationIndexTemplate) => {
+          const customMappings = transformsSpecifications
+            .get(destinationIndexTemplate.transformModuleId)
+            ?.get('mappings');
+          const customTemplateName = `${destinationIndexTemplate.installationName}${USER_SETTINGS_TEMPLATE_SUFFIX}`;
+          const packageTemplateName = `${destinationIndexTemplate.installationName}${PACKAGE_TEMPLATE_SUFFIX}`;
+          if (destinationIndexTemplate !== undefined || customMappings !== undefined) {
+            return installComponentAndIndexTemplateForDataStream({
+              esClient,
+              logger,
+              componentTemplates: {
+                ...(customMappings
+                  ? {
+                      [customTemplateName]: {
+                        template: {
+                          mappings: customMappings,
+                        },
+                        _meta: destinationIndexTemplate._meta,
+                      },
+                    }
+                  : {}),
+                ...(destinationIndexTemplate.template
+                  ? {
+                      [packageTemplateName]: {
+                        template: {
+                          settings: destinationIndexTemplate.template.settings,
+                          mappings: destinationIndexTemplate.template.mappings,
+                        },
+                        _meta: destinationIndexTemplate._meta,
+                      },
+                    }
+                  : {}),
+              },
+              indexTemplate: {
+                templateName: destinationIndexTemplate.installationName,
+                // @ts-expect-error Index template here should not contain data_stream property
+                // as this template is applied to only an index and not a data stream
+                indexTemplate: {
+                  template: { settings: undefined, mappings: undefined },
+                  priority: DEFAULT_TRANSFORM_TEMPLATES_PRIORITY,
+                  index_patterns: [
+                    transformsSpecifications
+                      .get(destinationIndexTemplate.transformModuleId)
+                      ?.get('destinationIndex').index,
+                  ],
+                  _meta: destinationIndexTemplate._meta,
+                  composed_of: [
+                    ...(customMappings ? [customTemplateName] : []),
+                    ...(destinationIndexTemplate.template ? [packageTemplateName] : []),
+                  ],
+                },
+              },
+            });
+          }
+        })
+        .filter((p) => p !== undefined)
+    );
+
+    // create destination indices
+    await Promise.all(
+      transforms.map(async (transform) => {
+        const index = transform.content.dest.index;
+        const pipelineId = transform.content.dest.pipeline;
+
+        return esClient.indices.create(
```

Comment:
Might be worth wrapping this in Fleet's `retryTransientEsErrors` helper as we do in other installation processes.

Reply:
Updated here [`5a98445` (#140046)](https://github.com/elastic/kibana/pull/140046/commits/5a98445cfc4c630186c503ad2cfca7893628b6a8)